system/Database/BasePreparedQuery.php

```diff
@@ -147,17 +147,26 @@ abstract public function _execute(array $data): bool;
     abstract public function _getResult();
 
     /**
-     * Explicitly closes the statement.
+     * Explicitly closes the prepared statement.
      */
-    public function close()
+    public function close(): bool
     {
-        if (! is_object($this->statement) || ! method_exists($this->statement, 'close')) {
-            return;
+        if (! isset($this->statement)) {
+            throw new BadMethodCallException('Cannot call close on a non-existing prepared statement.');
         }
 
-        $this->statement->close();
+        try {
+            return $this->_close();
+        } finally {
+            unset($this->statement);
```

Comment:
I just found out that unsetting a property removes it from the object at runtime, which may not be desired or when accessed accidentally after will error "Undefined property ...". I think this should instead be set to `null`.
```suggestion
            $this->statement = null;
```

Then, to satisfy phpstan, please add `|null` in the docblock of `$statement` in `BasePreparedQuery`.

Reply:
That should be ok.